src/cdk/schematics/testing/test-case-setup.ts

```diff
@@ -94,7 +93,10 @@ export async function createTestCaseSetup(migrationName: string, collectionPath:
   });
 
   const testAppTsconfigPath = 'projects/cdk-testing/tsconfig.app.json';
-  const testAppTsconfig = JSON.parse(appTree.readContent(testAppTsconfigPath));
+  // Parse TypeScript configuration files with JSON5 as they could contain comments or
+  // unquoted properties.
+  const testAppTsconfig =
+      parseJson(appTree.readContent(testAppTsconfigPath), JsonParseMode.Json5) as any;
```

Comment:
Nit that can be addressed later: I think TS has an interface for the config somewhere which we could reference here instead of `any`.

Reply:
I don't think there is such an interface. IIRC there is only `ts.CompilerOptions` and that deals only with the compiler options, and also relies on enum values. Let me know if you find something. That would definitely be ideal 😄

Reply:
In language service `ts.readConfigFile()` is used to parse JSON. It returns `any` though.

Reply:
https://github.com/microsoft/TypeScript/blob/d1ebf126d2bf6e381149a9d3ab29fb6c7076fea3/lib/tsserverlibrary.d.ts#L4477-L4484

Reply:
Yeah, that would be a good alternative. Might just that one when I come across it again. Looks like it would require us to handle parse diagnostics manually though.